src/components/Header.jsx

```diff
@@ -4,14 +4,44 @@ import { DiamondIcon } from '@/components/DiamondIcon'
 import Logo from 'public/images/logo.svg'
 import Image from 'next/image'
 import Link from 'next/link'
+import { useTheme } from 'next-themes'
+import { useState, useEffect } from 'react'
+import { MoonIcon, SunIcon } from '@heroicons/react/outline'
 
 export function Header() {
+  const { systemTheme, theme = 'dark', setTheme } = useTheme()
+  const [mounted, setMounted] = useState(false)
+
+  useEffect(() => setMounted(true), [])
+
+  const renderThemeChanger = () => {
+    if (!mounted) return null
+
+    const currentTheme = theme === 'system' ? systemTheme : theme
+
+    if (currentTheme === 'dark') {
```

Comment:
In order to remove this if statement that is duplicating code just do something like:

```c
  const icon = currentTheme === 'dark' ? <MoonIcon /> : <SunIcon />
```